src/common/MessageDispatcher.sol

```diff
@@ -600,6 +603,37 @@ contract MessageDispatcher is Auth, IMessageDispatcher {
         }
     }
 
+    /// @inheritdoc ISpokeMessageSender
+    function sendUpdateHubContract(
+        PoolId poolId,
+        ShareClassId scId,
+        bytes32 target,
+        bytes32 sender,
+        bytes calldata payload,
+        uint128 extraGasLimit,
+        address refund
+    ) external payable auth {
+        uint16 hubCentrifugeId = poolId.centrifugeId();
+
+        if (hubCentrifugeId == localCentrifugeId) {
+            hubContractUpdater.execute(poolId, scId, sender.toAddress(), target.toAddress(), payload);
```

Comment:
FYI, the order or sender and target does not correlate the message. I think we need to normalize those to avoid issues.